src/Handler.php

```diff
@@ -12,24 +12,85 @@ class Handler
     protected $client;
 
     /**
+     * The previously registered error handler as returned by the interpreter.
+     *
+     * @var callable|null
+     */
+    protected $previousErrorHandler;
+
+    /**
+     * The previously registered exception handler as returned by the interpreter.
+     *
+     * @var callable|null
+     */
+    protected $previousExceptionHandler;
+
+    /**
      * Register our exception handler.
      *
      * @param \Bugsnag\Client|string|null $client client instance or api key
+     * @param bool $callPrevious whether or not to call the previous handlers
      *
      * @return static
      */
-    public static function register($client = null)
+    public static function register($client = null, $callPrevious = true)
     {
         $handler = new static($client instanceof Client ? $client : Client::make($client));
 
-        set_error_handler([$handler, 'errorHandler']);
-        set_exception_handler([$handler, 'exceptionHandler']);
-        register_shutdown_function([$handler, 'shutdownHandler']);
+        $handler->registerErrorHandler($callPrevious);
+        $handler->registerExceptionHandler($callPrevious);
+        $handler->registerShutdownHandler();
 
         return $handler;
     }
 
     /**
+     * Register the bugsnag error handler and save the previous one
+     * (if it exists) to call later.
+     *
+     * @param bool $callPrevious whether or not to call the previous handler
+     *
+     * @return void
+     */
+    public function registerErrorHandler($callPrevious = true)
```

Comment:
Up to y'all. Setting the default to false will preserve the current broken behavior for new installations and create a bug that can lurk for a while.

This change will only break backwards compatibility people who

1. load their custom handler
1. call `Bugsnag::register`
1. re-register their custom handler on top of that, saving the values returned by `set_x_handler` and re-calling that

I'm guessing those who have already worked around this will have just called `Bugsnag\Handler::xHandler` directly, in which case `previousXHandler` will be `null` since `register` is never called.

I'd think calling the user's function twice is better than never. 🤷‍♂️